### PR TITLE
Improved performance of json rendering with `$`

### PR DESCRIPTION
**Pull request by onionhammer**
```
Current Standard Library
  time: 3.064078 s
New Json Renderer
  time: 2.634936 s
```

### CODE REVIEW

**Comment by dom96** [comment]
2 space indent please.

**Comment by dom96** [comment]
Also, use a new branch for your PRs next time please (saves us from problems if you have multiple PRs)

File: lib/pure/json.nim [hunk 1]
```diff
@@ -949,10 +949,50 @@ proc pretty*(node: JsonNode, indent = 2): string =
   result = ""
   toPretty(result, node, indent)
 
+proc toUgly*(result: var string, node: JsonNode) =
+  ## Converts `node` to its JSON Representation, without
+  ## regard for human readability.
```
Comment (dom96) [inline lib/pure/json.nim]: Mention in a comment here that this function has been written to make `$` faster.

**Comment by onionhammer** [comment]
Benchmark code

Code
-------
```nim
import json, times, strutils

let jsonData = json.parseFile("data.json")

when isMainModule:
    var output: string
    template bench(name: expr, op: stmt) =
        echo name
        let start = cpuTime()
        for i in 0.. 1000:
            op
        echo "  time: ", cpuTime() - start, " s"

    bench "New Json Renderer":
        output = toUgly(jsonData)

    bench "Current Standard Library":
        output = pretty(jsonData, 0)
```

```py
import json
import time

with open('data.json') as data_file:
    data = json.load(data_file)

def convert():
    result = ""
    for i in range(1000):
        result = str(data)


print "Python"
t1 = time.time()
convert()
t2 = time.time()
print "  time:", t2 - t1, "s"
```

Output
---------
```
New Json Renderer
  time: 2.511336 s
Current Standard Library
  time: 3.051935 s
Python
  time: 4.14301919937 s
```

File: lib/pure/json.nim [hunk 2]
```diff
@@ -949,10 +949,51 @@ proc pretty*(node: JsonNode, indent = 2): string =
...
+  of JInt:
+    result.add($node.num)
+  of JFloat:
+    result.add($node.fnum)
+  of JBool:
+    result.add(if node.bval: "true" else: "false")
+  of JNull:
+    result.add "null"
+
+proc toUgly*(node: JsonNode): string =
```
Comment (dom96) [inline lib/pure/json.nim]: There is no reason for this proc to exist as its equivalent to ``$``.

Comment (onionhammer) [inline lib/pure/json.nim]: True, I'll remove it with my next set of changes once you merge it :P

Comment (dom96) [inline lib/pure/json.nim]: No. Remove it from this PR.

Comment (onionhammer) [inline lib/pure/json.nim]: Can't edit it, the fork was deleted.

Comment (dom96) [inline lib/pure/json.nim]: Why was it deleted?

Comment (dom96) [inline lib/pure/json.nim]: It seems that I can't even manually merge it.

Comment (dom96) [inline lib/pure/json.nim]: In any case, I've merged it and fixed it myself.

**Pull request merged**